packages/components/src/components/validate/validate.component.ts

```diff
@@ -0,0 +1,285 @@
+import type { CSSResultGroup } from 'lit';
+import { html } from 'lit';
+import { property, queryAssignedElements, state } from 'lit/decorators.js';
+import componentStyles from '../../styles/component.styles.js';
+import SynergyElement from '../../internal/synergy-element.js';
+import { watch } from '../../internal/watch.js';
+import SynAlert from '../alert/alert.component.js';
+import {
+  getEventNameForElement,
+  isBlurEvent,
+  isInvalidEvent,
+  normalizeEventAttribute,
+} from './utility.js';
+import styles from './validate.styles.js';
+
+/**
+ * @summary Validate is a helper that may be used to wrap
+ * synergy input fields and forms to provide validation message.
+ *
+ * @dependency syn-alert
+ *
+ * @slot - The input field or form element to be validated.
+ * Avoid slotting in more than one element, as subsequent ones will be ignored.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart input-wrapper - The container that wraps the input field.
+ * @csspart alert - The syn-alert that is shown when the variant is set to "inline".
+ * @csspart alert__base - The container that wraps the alert.
+ * @csspart alert__message - The container that wraps the alert message.
+ * @csspart alert__icon - The container that wraps the alert icon.
+ */
+export default class SynValidate extends SynergyElement {
+  static styles: CSSResultGroup = [componentStyles, styles];
+
+  static dependencies = {
+    'syn-alert': SynAlert,
+  };
+
+  controller = new AbortController();
+
+  @queryAssignedElements() private slottedChildren: HTMLElement[];
+
+  @state() validationMessage = '';
+
+  /**
+   * The variant that should be used to show validation alerts.
+   *
+   * The following variants are supported:
+   * - **native** (default): Uses the native browser validation, usually a browser tooltip.
+   * - **inline**: Show the validation message underneath the element, using a `<syn-alert>`
+   */
+  @property({ reflect: true }) variant: 'native' | 'inline' = 'native';
+
+  /** Do not show the error icon when using the inline variant validation */
+  @property({ attribute: 'hide-icon', reflect: true, type: Boolean }) hideIcon = false;
+
+  /**
+   * Defines the events that trigger the validation.
+   * `invalid` will always automatically be included.
```

Comment:
maybe its good to link to msdn (https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/invalid_event) so people know it's a standard html feature ...

Reply:
This is now directly linked and available in Storybook, as well as in type information.